creusot/src/clone_map.rs

```diff
@@ -48,72 +54,61 @@ impl PreludeModule {
     }
 }
 
+#[derive(Clone)]
 pub struct CloneMap<'tcx> {
     tcx: TyCtxt<'tcx>,
     prelude: IndexSet<PreludeModule>,
     names: IndexMap<(DefId, SubstsRef<'tcx>), CloneInfo>,
     count: usize,
+    item_type: ItemType,
 }
 
 #[derive(Clone)]
-struct CloneInfo {
+pub struct CloneInfo {
     name: String,
-    projs: Vec<CloneSubst>,
     hidden: bool,
 }
 
 impl CloneInfo {
     fn from_name(name: String) -> Self {
-        CloneInfo { name, projs: Vec::new(), hidden: false }
+        CloneInfo { name, hidden: false }
     }
 
     fn hidden(name: String) -> Self {
-        CloneInfo { name, projs: Vec::new(), hidden: true }
+        CloneInfo { name, hidden: true }
     }
-}
 
-impl<'tcx> CloneMap<'tcx> {
-    pub fn new(tcx: TyCtxt<'tcx>) -> Self {
-        let names = IndexMap::new();
-        CloneMap { tcx, names, count: 0, prelude: IndexSet::new() }
+    // TODO: When traits stop holding all functions we can remove the last two arguments
+    pub fn qname(&self, tcx: TyCtxt, def_id: DefId) -> QName {
```

Comment:
The rather ugly type of this function will require #30 to be addressed first before it can be properly fixed.

Reply:
If each clone only introduces a single symbol, we can cache that symbol name in the `CloneInfo`, I believe that the solution to #30 will require splitting up traits / implementations into one module per symbol so this would address this issue.